src/string.cr

```diff
@@ -1743,6 +1743,21 @@ class String
     end
   end
 
+  # Relturns a string where all strings in the given hash are replaced.
+  # by the corresponding *hash* values.
+  #
+  # ```
+  # "hello".gsub({"e" => 'a', "l" => 'd'}) # "haddo"
+  # ```
+  def gsub(hash : Hash(String, _))
+    string = self
+    hash.each do |key, value|
+      string = string.gsub(key, value)
```

Comment:
I was curious to see what the difference between the additional `String#gsub(Hash(String, _))` versus the already existing `String#gsub(Hash(Char, _))`

```crystal
require "benchmark"

string  = "Apples " * 10000
hash    = { "p" => 'a', "l" => 's' }
inverse = hash.invert

Benchmark.bm do |x|
  x.report("Gsub with Hash String => Char 2000 times") do
    2000.times do
      string.gsub(hash)
    end
  end

  x.report("Gsub with Hash Char => String 2000 times") do
    2000.times do
      string.gsub(inverse)
    end
  end
end
```

I have run this four times and so far my implementation seems faster on my MacBook. Keep in mind I am new to crystal maybe there is a better way to do this benchmark.

```
                                               user     system      total        real
Gsub with Hash String => Char 2000 times   1.400000   0.270000   1.670000 (  1.499639)
Gsub with Hash Char => String 2000 times   1.620000   0.160000   1.780000 (  1.669893)
```

```
                                               user     system      total        real
Gsub with Hash String => Char 2000 times   1.390000   0.270000   1.660000 (  1.482408)
Gsub with Hash Char => String 2000 times   1.610000   0.150000   1.760000 (  1.669062)
```

```
                                               user     system      total        real
Gsub with Hash String => Char 2000 times   1.380000   0.270000   1.650000 (  1.471932)
Gsub with Hash Char => String 2000 times   1.620000   0.150000   1.770000 (  1.677077)
```

```
                                               user     system      total        real
Gsub with Hash String => Char 2000 times   1.390000   0.270000   1.660000 (  1.482940)
Gsub with Hash Char => String 2000 times   1.610000   0.150000   1.760000 (  1.668073)
```